server/server.go

```diff
@@ -851,7 +861,114 @@ func (s *Server) handleExit(ctx context.Context, event fsnotify.Event) {
 	if err := s.ContainerStateToDisk(ctx, c); err != nil {
 		log.Warnf(ctx, "Unable to write %s %s state to disk: %v", resource, c.ID(), err)
 	}
+	if s.config.EventedPLEG {
+		err := s.generateCRIEvent(ctx, c, types.ContainerEventType_CONTAINER_STOPPED_EVENT)
+		if err != nil {
+			log.Warnf(ctx, "Unable to generate event %s for container %s due to err %s", types.ContainerEventType_CONTAINER_STOPPED_EVENT, c.ID(), err)
+		}
+	}
 	if err := os.Remove(event.Name); err != nil {
 		log.Warnf(ctx, "Failed to remove exit file: %v", err)
 	}
 }
+
+func (s *Server) getSandboxStatuses(ctx context.Context, sandboxId string) (*types.PodSandboxStatus, error) {
+	sandboxStatusRequest := &types.PodSandboxStatusRequest{PodSandboxId: sandboxId}
+	sandboxStatus, err := s.PodSandboxStatus(ctx, sandboxStatusRequest)
+
+	if IsNotFound(err) {
+		return nil, err
+	}
+
+	if err != nil {
+		return nil, err
```

Comment:
Let's wrap errors to be trace-able:
```suggestion
		return nil, fmt.Errorf("get pod sandbox statuses: %w", err)
```